docs(readme): Linux installers, nixos docs, readme refactor and gruvb…

iruzo opened the pull request:
## Proposed Changes

  - Trying to reduce size of README and improving readability.
  - Added one click installers for unix/linux as POSIX sh scripts.
  - Documented NixOS installation.
  - Added gruvbox theme (preview webp is still needed).

andreasgrafen commented:
Ayo! That's a lot. **Big** thank you for this! <3

---
### Readme

As you have mentioned on Discord; some of these changes are highly objective. Personally I find the readme changes more confusing to read through. Things are also a little bit out of order with the previews just kinda hanging there without being attached to a specific section. o:

---
### Gruvbox

The Gruvbox theme is lovely actually! I'll for sure add that one into the mix for the colour schemes. c:

---
### Installer

I also quite like the addition of the installation script, tho if I read that right, there are some issues with it.

First up, the order of changing to the config dir, and removing a potentially pre-existing chrome folder could be swapped to make things a little less convoluted. The default directory may also not always be caught by the regex, I believe? But I'd say that can be omitted.
What I think should def. be there is some sort of warning and check to confirm that an existing chrome folder should be deleted as this might cause unwanted data loss!

Furthermore I'd personally like to have just one installer script that would take arguments for the desired colour scheme to declutter the whole thing a little more. (And maybe even support macOS as well)

I'm happy to look into the changes for the installer script and put them into a separate branch in order for us to make things nice. ^—^

iruzo commented:
Hello @andreasgrafen !, check the last changes I made pls.

I changed the installer script, so you can pass the theme as parameter (in order to do that, I changed the themes directory from `integrations/theme/theme.css` to 'integrations/theme.css').

I also changed the README to move the theming preview to the customization section and removed the emojis, if this doesn't look good enough, I can restore the README to default and just add the doc for gruvbox and the installer.

42willow commented:
Hi @iruzo  👋, sorry for the wait - I’m picking up maintenance of this theme (at least for now, we’ll see how I go 😅)

### gruvbox

thanks for the gruvbox theme! I’ll be sure to merge it when I have some free time - it would also be nice to include the light mode variant but I can get to that

### installer

the installer looks good :) it will definitely speed up setup - in my opinion it would also be nice for the script to handle the downloading of the theme too, but this may be subjective

### nixos

as for the nixos docs, I think it’d be nice to offer a home manager module - similar to how textfox does it, which will also increase qol for configuration changes

### readme

I personally agree with @unseen-ninja on that the current layout of the readme is more accessible - but having all the documentation as one big slab of text can be a bit tricky to navigate (especially without a table of contents)
maybe we could look into using vitepress or starlight for this


thanks :)

42willow commented:
Could possibly use https://github.com/ryanccn/nyoom for this, it's cross platform too!

42willow commented:
No problem! I've been a bit busy too and it's about to get busier.

Very good point, I agree with you on preferring an installation script over nyoom - especially since it isn't in pkg managers yet (as far as i know)... 

Unfortunately I think modifying and maintaining a fork of AddWater for Cascade might be out of the scope of my knowledge (and it'd be Linux only...) - but if someone else would like to pick it up they could

I made some changes to the script a while back, not sure if they're any good but here they are for reference anyway:

```bash
#!/bin/sh

PROFILES=($(grep 'Path=' "$HOME/.mozilla/firefox/profiles.ini" | cut -d= -f2))

# ask the user to select a profile if there are multiple
if [ "${#PROFILES[@]}" -eq 1 ]; then
  SELECTED_PROFILE=${PROFILES[0]}
elif [ "${#PROFILES[@]}" -eq 0 ]; then
  echo "No profiles found. Operation cancelled."
  exit 1
else
  for i in "${!PROFILES[@]}"; do
    echo "$i) ${PROFILES[$i]}"
  done

  read -p "Select a profile: " profile_number

  if ! [[ "$profile_number" =~ ^[0-9]+$ ]] || [ "$profile_number" -ge "${#PROFILES[@]}" ]; then
    echo "Invalid selection. Operation cancelled."
    exit 1
  fi

  SELECTED_PROFILE=${PROFILES[$profile_number]}
fi

echo "Selected profile: $SELECTED_PROFILE"
echo "Path: $HOME/.mozilla/firefox/$SELECTED_PROFILE/chrome"

# Confirm the selected profile
read -p "Is this the correct profile? (Y/n): " confirm
if [ "$confirm" != "y" ] && [ "$confirm" != "" ]; then
  echo "Operation cancelled."
  exit 1
fi

rm "$HOME/.mozilla/firefox/$SELECTED_PROFILE/chrome" -rf

TEMP_DIR=$(mktemp -d)
echo "temporary dir created at $TEMP_DIR"

git clone https://github.com/cascadefox/cascade "$TEMP_DIR/cascade"
echo "repository cloned to $TEMP_DIR/cascade"

mv "$TEMP_DIR/cascade/chrome" "$HOME/.mozilla/firefox/$SELECTED_PROFILE/"
echo "chrome directory moved to Firefox profile"


# if [ "$#" -gt 0 ]; then
#   mv "$TEMP_DIR/cascade/integrations/cascade-$1.css" "$HOME/.mozilla/firefox/$SELECTED_PROFILE/chrome/includes/cascade-colours.css"
#   echo "Theme configured: cascade-$1.css"
# fi

# Clean up the temporary directory
rm -rf "$TEMP_DIR"
echo "Temporary directory cleaned up"

# Change toolkit.legacyUserProfileCustomizations.stylesheets
if [ ! -e "$HOME/.mozilla/firefox/$SELECTED_PROFILE/prefs.js" ]; then
  echo "File 'prefs.js' does not exist."
  exit 1
fi

if grep -q "user_pref(\"toolkit.legacyUserProfileCustomizations.stylesheets\", true);" "$HOME/.mozilla/firefox/$SELECTED_PROFILE/prefs.js"; then
  echo "legacyUserProfileCustomizations is already set to true"
else
  echo "user_pref(\"toolkit.legacyUserProfileCustomizations.stylesheets\", true);" >> "$HOME/.mozilla/firefox/$SELECTED_PROFILE/prefs.js"
  echo "legacyUserProfileCustomizations set to true"
fi

echo "Installation complete."
```
